### PR TITLE
add support to set repository_version if init restic repository

### PR DESCRIPTION
**Pull request by unclesamwk**
The newer versions of restic use a new repository format with new features. In legacy environments it may be useful to support both version of repository formats.

### CODE REVIEW

**Comment by mtlynch** [comment]
Thanks for contributing! I didn't realize there's a new format.

Can you [document this new flag](https://github.com/mtlynch/resticpy/blob/master/docs/index.md#init) and add [a unit test](https://github.com/mtlynch/resticpy/blob/master/restic/internal/init_test.py) to exercise the new behavior?

**Comment by unclesamwk** [comment]
Ive update the pullrequest. Please check the unit test, i dont know it is the right way

File: restic/internal/init_test.py [hunk 1]
```diff
@@ -48,3 +48,19 @@ def test_init_with_secondary_repository(self, mock_execute):
...
+""".strip()
+
+        repository_id = restic.init(repository_version=1)
+
+        self.assertEqual(
+            'd0c84a66bffea61b4cbb88c39cea742127699b3f3af71127b68edcc142edff48',
+            repository_id)
+        mock_execute.assert_called_with([
+            'restic', '--json', 'init', '--repository-version', '1',
+            's3:https://some.backend.com/mybucket'
```
Comment (mtlynch) [inline restic/internal/init_test.py:65]: This line seems to be accidentally copy/pasted, and it's breaking the test.

**Review by mtlynch — CHANGES_REQUESTED** [review]

File: restic/internal/init_test.py (hunk 1 above)
Comment (unclesamwk) [reply restic/internal/init_test.py]: Now its fixed!

**Comment by mtlynch** [comment]
Thanks for adding this, @unclesamwk!

**Pull request merged**